Cython/Compiler/PyrexTypes.py

```diff
@@ -4019,6 +4021,73 @@ def check_nullary_constructor(self, pos, msg="stack allocated"):
         if constructor is not None and best_match([], constructor.all_alternatives()) is None:
             error(pos, "C++ class must have a nullary constructor to be %s" % msg)
 
+class CppScopedEnumType(CType):
+    # name    string
+    # cname   string
+
+    is_cpp_enum = True
+
+    def __init__(self, name, cname, underlying_type, namespace=None):
+        self.name = name
+        self.cname = cname
+        self.values = []
+        self.underlying_type = underlying_type
+        self.namespace = namespace
+
+    def __str__(self):
+        return self.name
+
+    def declaration_code(self, entity_code,
+                        for_display=0, dll_linkage=None, pyrex=0):
+        if pyrex or for_display:
+            base_code = self.name
+        else:
+            if self.namespace:
+                base_code = "%s::%s" % (
+                    self.namespace.empty_declaration_code(),
+                    self.cname
+                )
+            else:
+                base_code = "enum %s" % self.cname
+            base_code = public_decl(base_code, dll_linkage)
+        return self.base_declaration_code(base_code, entity_code)
```

Comment:
The name `base_code` suggests that it has something to do with the base type, but the opposite is the case. Something like `type_code` or `type_name` sounds more appropriate.

Reply:
Fixed (used `type_name` instead)